zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/metrics/StreamProcessorMetrics.java

```diff
@@ -64,10 +65,33 @@ public final class StreamProcessorMetrics {
           .help("Time taken for startup and recovery of stream processor (in ms)")
           .labelNames(LABEL_NAME_PARTITION)
           .register();
+
+  private static final Gauge PROCESSOR_STATE =
+      Gauge.build()
+          .namespace(NAMESPACE)
+          .name("stream_processor_state")
+          .help("Describes the state of the stream processor, namely if it is active or paused.")
+          .labelNames(LABEL_NAME_PARTITION)
+          .register();
   private final String partitionIdLabel;
+  private final Gauge.Child processorState;
 
-  public StreamProcessorMetrics(final int partitionId) {
+  public StreamProcessorMetrics(final int partitionId, final Phase streamProcessorPhase) {
     partitionIdLabel = String.valueOf(partitionId);
+    processorState = PROCESSOR_STATE.labels(partitionIdLabel);
+    if (streamProcessorPhase.equals(Phase.PAUSED)) {
+      setStreamProcessorPaused();
+    } else {
+      setStreamProcessorActive();
+    }
+  }
+
+  public void setStreamProcessorActive() {
+    processorState.set(0);
+  }
+
+  public void setStreamProcessorPaused() {
+    processorState.set(1);
```

Comment:
:question: Wouldn't it be use ful to include all StreamProcessor phase in the metric 

```
    INITIAL,
    REPLAY,
    PROCESSING,
    FAILED,
    PAUSED,
```
This would improve the observability when stream processor is failed.